--- confbridge.h.orig   2013-10-09 02:14:14.000000000 +0600
+++ confbridge.h        2013-12-25 17:19:55.000000000 +0600
@@ -240,6 +240,7 @@
        unsigned int suspended_moh;                  /*!< Count of active suspended MOH actions. */
        unsigned int kicked:1;                       /*!< User has been kicked from the conference */
        unsigned int playing_moh:1;                  /*!< MOH is currently being played to the user */
+       unsigned int talking:1;                      /* talking !!! av added */
        AST_LIST_HEAD_NOLOCK(, post_join_action) post_join_list; /*!< List of sounds to play after joining */;
        AST_LIST_ENTRY(conference_bridge_user) list; /*!< Linked list information */
 };
